internal/gatewayapi/testdata/envoyproxy-tls-settings.in.yaml

```diff
@@ -0,0 +1,133 @@
+gateways:
```

Comment:
Can we also add MTLS + TLS Settings tests for ext-auth and ext-proc, demonstrating how this applied to these sorts of backends?